src/FedoraApi.php

```diff
@@ -230,4 +230,85 @@ public function deleteResource(
             $options
         );
     }
+
+    /**
+     * Saves RDF in Fedora.
+     *
+     * @param EasyRdf_Resource  $graph          Graph to save
+     * @param string            $uri            Resource URI
+     * @param array             $headers        HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function saveGraph(
+        \EasyRdf_Graph $graph,
+        $uri = '',
+        $headers = []
+    ) {
+        // Serialze the rdf.
+        $turtle = $graph->serialise('turtle');
+
+        // Checksum it.
+        $checksum_value = sha1($turtle);
+
+        // Set headers.
+        $headers['Content-Type'] = 'text/turtle';
+        $headers['digest'] = 'sha1=' . $checksum_value;
+
+        $options['headers'] = $headers;
+        $options['body'] = $turtle;
+
+        // Save it.
+        return $this->saveResource($uri, $turtle);
```

Comment:
You don't need the `$options` array at all, and you need to pass the headers in to `saveResource()`